scripts: add new-backend migration script (#4824)
ElektraInitiative/libelektra: doc/news/_preparation_next_release.md

The structure of `system:/elektra/mountpoints` changed as well.
Take a look at the [new docs](../dev/mountpoints.md), if you need to know details.

#### Updating config

> > **Warning**: BACK UP YOUR CONFIG FILES BEFORE UPDATING!

Reviewer: Can you show here what should be backed up (or make the Python tool back it up).

Author: In theory, you don't need to back up everything. The script doesn't even overwrite anything unless you explicitly tell it to.

I practice, if you use the `--output` option, you may want to back up the file you specify before overwriting it. But that should be pretty obvious. 

I just put the warning here to be on the safe side.

To update your existing `system:/elektra/mountpoints` data you can use the [migration script](/scripts/migrate-mountpoints.py).

> > **Note**: To run the script you must have Elektra and the Python binding installed.

Reviewer: Python >= 3.10 seemingly also needs to be installed.

> The script uses the Python binding to manipulate `Key`s and `KeySet`s, but it does not the `kdb` CLI tool, or the `KDB` API.
> It is safe to run this script before, or after you update your Elektra installation.

By default, the script loads the file `/etc/kdb/elektra.ecf`.
If you changed where `system:/elektra/mountpoints` is stored, you can provide an alternative path:

```
./migrate-mountpoint.py /path/to/your/mountpoints/config.file
```

> > **Note**: Because the script does not use the `KDB` API it only works, if the mountpoints config file uses the default `dump` format.

Reviewer: You could use `kdb convert` before?

Author: Yes, but it'll probably only work, if you run `kdb convert` before updating Elektra. After updating `kdb` probably crashes because it can't load it's own config.

The script will read the old mountpoint configuration from the given file.
It will convert the configuration and print the new version to `stdout`.

You can inspect the output to make sure, everything is in order.
When you are ready to commit the changes, you can manually edit the config file, or use:

```

> ./migrate-mountpoint.py --output /etc/kdb/elektra.ecf

Author: ```suggestion
./migrate-mountpoint.py --output=/etc/kdb/elektra.ecf /etc/kdb/elektra.ecf
```

The structure of `system:/elektra/mountpoints` changed as well.
Take a look at the [new docs](../dev/mountpoints.md), if you need to know details.

#### Updating config

> **Warning**: BACK UP YOUR CONFIG FILES BEFORE UPDATING!
> We recommend making a backup of the file printed by `kdb file system:/elektra/mountpoints` before updating your installation.
> In the unlikely case that the migration script fails, you can still use the information from the backup to manually recreate your mountpoints.

To update your existing `system:/elektra/mountpoints` data you can use the [migration script](/scripts/migrate-mountpoints.py).

> > **Note**: To run the script you must have Elektra and Python the Python binding installed.
> > The minimum Python version with which the script works is 3.7.
> > The script uses the Python binding to manipulate `Key`s and `KeySet`s, but it does not the `kdb` CLI tool, or the `KDB` API.

Reviewer 2: ```suggestion
> **Note**: To run the script you must have Elektra, Python (>= 3.7) and the Python binding installed.
> The script uses the Python binding to manipulate `Key`s and `KeySet`s, but it does not use the `kdb` CLI tool, or the `KDB` API.
```
